dags/glam_fenix.py

```diff
@@ -84,17 +88,63 @@
     sum(LOGICAL_MAPPING.values(), [])
 )
 for product in final_products:
-    query = generate_and_run_glean_query(
-        task_id=f"incremental_{product}",
+    func = partial(
+        generate_and_run_glean_task,
         product=product,
         destination_project_id=PROJECT,
         env_vars=dict(STAGE="incremental"),
         dag=dag,
     )
+    view, init, query = [
+        partial(func, task_type=task_type) for task_type in ["view", "init", "query"]
+    ]
+
+    # stage 1 - incremental
+
+    cdha = view(task_name=f"{product}__view_clients_daily_histogram_aggregates_v1")
+    cdsa = view(task_name=f"{product}__view_clients_daily_scalar_aggregates_v1")
+    lv = view(task_name=f"{product}__latest_versions_v1")
+    # only the scalar aggregates are downstream of latest versions
+    cdsa >> lv
+
     # get the dependencies for the logical mapping, or just pass through the
     # daily query unmodified
     for dependency in LOGICAL_MAPPING.get(product, [product]):
-        mapping[dependency] >> query
+        mapping[dependency] >> cdsa
+        mapping[dependency] >> cdha
+
+    csa_init = init(task_name=f"{product}__clients_scalar_aggregates_v1")
+    csa = query(task_name=f"{product}__clients_scalar_aggregates_v1")
+
+    cdsa >> csa_init
+    lv >> csa_init
```

Comment:
If `cdsa >> lv` and `lv >> csa_init`, do we need `cdsa >> csa_init`?

Reply:
Technically no since the dependency is transitive, but its good practice to lay out the dependencies here. Airflow can run `lv` and `csa_init` at the same time here, instead of one after the other.